### PR TITLE
Sdem/remove component adaptor

### PR DESCRIPTION
**Pull request by maceligueta**
**Description**
Separate PR for #7511. 
Please @GuillermoCasas , take a look at these changes, I might have broken something. I did not run the tests yet.

**Changelog**
- Simple DEM removals of useless includes
- I tried to change the adaptor to a simple 'Variable<double>'. I am not sure if this was the expected change but at least it compiles.

### CODE REVIEW

File: applications/SwimmingDEMApplication/custom_python/add_custom_utilities_to_python.cpp [hunk 1]
```diff
@@ -329,7 +329,7 @@ void  AddCustomUtilitiesToPython(pybind11::module& m){
         .def(py::init<SpaceTimeSet::Pointer, VelocityField::Pointer, const double, const double >())
         ;
 
-    typedef void (CustomFunctionsCalculator<3>::*CopyValuesScalar)(ModelPart&, const VariableComponent<VectorComponentAdaptor<array_1d<double, 3> > >&, const VariableComponent<VectorComponentAdaptor<array_1d<double, 3> > >&);
+    typedef void (CustomFunctionsCalculator<3>::*CopyValuesScalar)(ModelPart&, const Variable<double>&, const Variable<double>&);
```
Comment (philbucher) [inline applications/SwimmingDEMApplication/custom_python/add_custom_utilities_to_python.cpp:332]: yes this is correct :+1:

**Review by philbucher — APPROVED** [review]
ping @maceligueta @GuillermoCasas this blocks from removing LOTS of deprecation warnings from the core

**Comment by maceligueta** [comment]
done, @philbucher

**Comment by philbucher** [comment]
thanks!

**Pull request merged**